### PR TITLE
[PORT] Replaces travis with github-actions

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

ports
tgstation/tgstation#54668
tgstation/tgstation#54833
tgstation/tgstation#54780
tgstation/tgstation#54805
tgstation/tgstation#53669
tgstation/tgstation#52529

## Why It's Good For The Game

Github-actions is faster and more consistent

### CODE REVIEW

**Comment by utilbird** [comment]
Godspeed, travman o7

**Comment by TeomanTheGreat** [comment]
Rip Travis may he rest in peace

**Comment by Crossedfall** [comment]
Add the `workflow_dispatch` trigger so we can run tests on-demand.

**Comment by Crossedfall** [comment]
> Github-actions is faster and more consistent

image

**Comment by jupyterkat** [comment]
run_all_tests is not working correctly for some reason

**Comment by MCterra10** [comment]
github actions is not ready for this.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        run: pwsh tools/ci/build.ps1
+      - name: Create artifact
+        run: |
+          md deploy
+          bash tools/deploy.sh ./deploy
+      - name: Deploy artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: deploy
+          path: deploy
```
Comment (Crossedfall) [inline .github/workflows/test.yml]:
```suggestion
```

Creating deploy artifacts from external forks is a terrible idea. It means that users could introduce malicious code into their PR and then GitHub would compile that into an executable deploy package that is tied to our repository.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+name: Run tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - master
```
Comment (Crossedfall) [inline .github/workflows/test.yml]:
```suggestion
  push:
    paths-ignore:
  	- 'html/changelogs/**'
    branches:
    - master
```

We don't want it to run for CL pushes

**Review by Crossedfall — CHANGES_REQUESTED** [review]

**Comment by Crossedfall** [comment]
Alright, lets give this a go.

**Review by Crossedfall — APPROVED** [review]

**Pull request merged**